src/core/hle/service/soc_u.cpp

```diff
@@ -320,6 +328,36 @@ union CTRSockAddr {
     }
 };
 
+struct CTRAddrInfo {
+    s32 ai_flags;
+    s32 ai_family;
+    s32 ai_socktype;
+    s32 ai_protocol;
+    s32 ai_addrlen;
+    char ai_canonname[256];
+    CTRSockAddr ai_addr;
+
+    /// Converts a platform-specific addrinfo to a 3ds addrinfo.
+    static CTRAddrInfo FromPlatform(const addrinfo& addr) {
+        CTRAddrInfo ctr_addr;
```

Comment:
```suggestion
        CTRAddrInfo ctr_addr{};
```

Is an equivalent that doesn't require memset (yes all padding bits are zeroed out as well).